### PR TITLE
Make esm rollup compatible

### PR DESCRIPTION
**Pull request by donaldinho**
As detailed in https://github.com/ldthomas/apg-js/issues/13 a number of people are experiencing issues when using this library as a dependency of other packages. The issue arises due to this package being a commonjs module that is not able to be correctly converted to an es module when bundled for the browser. 

This PR takes the references to the global this object and uses the exports object instead. This does not change the functionality for this commonjs module as referencing the members of the exports object is valid in commonjs as long as the member has been previously defined. This is the case for all 3 src files contained in this PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,10 +44,13 @@
   },
   "homepage": "https://github.com/ldthomas/apg-js#readme",
   "devDependencies": {
+    "browserify": "17.0.0",
+    "minify": "10.2.0",
```
Comment (benmccann) [inline package.json]: these should probably be inserted in alphabetical order

**Review by benmccann — APPROVED** [review]
these look like good changes. I'd love to see it go even a step further and migrate the source to ESM if @ldthomas is amenable

File: package.json (hunk 1 above)
Comment (donaldinho) [reply package.json]: pushed an update to sort correctly but for some reason the PR is not reflecting this

**Comment by ldthomas** [comment]
Please see comments in issue #13. I do plan to get this taken care of when able. I'm hoping I can get to it by Aug 1. Thanks for you input and comments.

**Comment by jwahdatehagh** [comment]
Tested this in my vite based setup and works. Thank you!

(Using the apg via SIWE)

For anyone needing a quick fix – just install the package from github

`"apg-js": "git+https://github.com/runic-inc/apg-js#8bb92ff0780819170fee7e756d26e0562e442983"`

and set your resolutions:

```json
"resolutions": {
    "apg-js": "git+https://github.com/runic-inc/apg-js#8bb92ff0780819170fee7e756d26e0562e442983",
    "siwe/**/apg-js": "git+https://github.com/runic-inc/apg-js#8bb92ff0780819170fee7e756d26e0562e442983",
    "siwe-parser/**/apg-js": "git+https://github.com/runic-inc/apg-js#8bb92ff0780819170fee7e756d26e0562e442983"
  }
  ```

**Comment by ldthomas** [comment]
Both the "exports" and "dependencies" issues suggested here are included in the release of `apg-js` version 4.2.0.

**Pull request closed without merging**